test/new_tests/test_free_threading.py

```diff
@@ -0,0 +1,40 @@
+import threading
+import pytest
+import aerospike
+
+
+@pytest.mark.usefixtures("as_connection")
+class TestFreeThreading:
+    def test_unsafe(self):
+        key = ("test", "demo", 1)
+        BIN_VALUE = 1
+        self.as_connection.put(key, bins={"a": BIN_VALUE})
+
+        bin_value_sum = 0
+
+        THREAD_COUNT = 10
+        barrier = threading.Barrier(parties=THREAD_COUNT)
+        config = {"hosts": [("127.0.0.1", 3000)]}
+        lock = threading.Lock()
+
+        def read_bin():
+            nonlocal barrier, config, lock
+            barrier.wait()
+            client = aerospike.client(config)
+            _, _, bins = client.get(key)
+            nonlocal bin_value_sum
+            lock.acquire()
+            bin_value_sum += bins["a"]
+            lock.release()
+
+        workers = []
+        for _ in range(THREAD_COUNT):
+            workers.append(threading.Thread(target=read_bin))
+
+        for worker in workers:
+            worker.start()
+
+        for worker in workers:
+            worker.join()
+
+        assert bin_value_sum == THREAD_COUNT * BIN_VALUE
```

Comment:
I would also report out to a log the timing so you can calculate and report out basic throughput numbers.  You can use this to sanity check to see if you get better numbers with a large transaction count, multiple threads running with and without GIL.

Reply:
Maybe it would be good to have both performance and data race testing?